tests/unit/jobserver/management/commands/test_check_rap_api_status.py

```diff
@@ -28,3 +71,133 @@ def fake_backend_status():
+    }
+    # no stats entry should be created
+    assert backend.stats.count() == 0
+
+
+def test_update_backend_state_existing_url(patch_backend_status_api_call):
+    backend = BackendFactory(name="test")
+    patch_backend_status_api_call(backend.name)
+
+    call_command("check_rap_api_status")
+    backend.refresh_from_db()
+
+    assert backend.rap_api_state == {
+        "name": backend.name,
+        "last_seen": "2025-08-12T06:57:43.039078Z",
+        "paused": {"status": "off", "since": "2025-08-12T14:33:57.413881Z"},
+        "db_maintenance": {"status": "off", "since": None, "type": None},
+    }
+    # check there's only one Stats for backend
+    assert backend.stats.count() == 1
+    assert backend.stats.first().url == "http://example.com/rap/"
+
+
+def test_update_backend_state_new_url(patch_backend_status_api_call, settings):
+    backend = BackendFactory(name="test")
+
+    patch_backend_status_api_call(backend.name)
+    call_command("check_rap_api_status")
+
+    backend.refresh_from_db()
+
+    assert backend.stats.count() == 1
+
+    # patch a new url path using the settings fixture
+    settings.RAP_API_BASE_URL = "http://example.com/rap/new"
+
+    patch_backend_status_api_call(backend.name)
+    call_command("check_rap_api_status")
+
+    backend.refresh_from_db()
+
+    # check there are now two Stats for backend
+    assert backend.stats.count() == 2
+    assert backend.stats.last().url == "http://example.com/rap/new"
+
+
+def test_update_backend_state_multiple_backends(monkeypatch):
+    backend1 = BackendFactory(name="test1")
+    backend2 = BackendFactory(name="test2")
+
+    test_response_body = {
+        "backends": [
+            {
+                "name": f"{backend1.name}",
+                "last_seen": "2025-08-12T06:57:43.039078Z",
+                "paused": {
+                    "status": "off",
+                    "since": "2025-08-12T14:33:57.413881Z",
+                },
+                "db_maintenance": {
+                    "status": "off",
+                    "since": None,
+                    "type": None,
+                },
+            },
+            {
+                "name": f"{backend2.name}",
+                "last_seen": "2025-08-12T06:57:43.039078Z",
+                "paused": {
+                    "status": "off",
+                    "since": "2025-08-12T14:33:57.413881Z",
+                },
+                "db_maintenance": {
+                    "status": "on",
+                    "since": "2025-08-12T14:33:57.413881Z",
+                    "type": "scheduled",
+                },
+            },
+        ]
+    }
+
+    monkeypatch.setattr(
+        "jobserver.rap_api.backend_status",
+        lambda: test_response_body,
+    )
+
+    call_command("check_rap_api_status")
+
+    backend1.refresh_from_db()
+
+    assert backend1.rap_api_state == test_response_body["backends"][0]
+    assert backend1.last_seen_at == datetime.datetime.fromisoformat(
```

Comment:
It might be worth adding an extra assert to check that `last_seen_maintenance_mode` is also None, since you're now explicitly setting it.